pkg/executor/aggregate/agg_hash_partial_worker.go

```diff
@@ -182,3 +288,100 @@ func (w *HashAggPartialWorker) shuffleIntermData(finalConcurrency int) {
 		w.outputChs[i] <- &w.partialResultsMap[i]
 	}
 }
+
+func (w *HashAggPartialWorker) prepareForSpill() {
+	if !w.isSpillPrepared {
+		w.tmpChksForSpill = make([]*chunk.Chunk, spilledPartitionNum)
+		w.spilledChunksIO = make([]*chunk.DataInDiskByChunks, spilledPartitionNum)
+		for i := 0; i < spilledPartitionNum; i++ {
+			w.tmpChksForSpill[i] = w.spillHelper.getNewSpillChunkFunc()
+			w.spilledChunksIO[i] = chunk.NewDataInDiskByChunks(w.spillHelper.spillChunkFieldTypes)
+			if w.spillHelper.diskTracker != nil {
+				w.spilledChunksIO[i].GetDiskTracker().AttachTo(w.spillHelper.diskTracker)
+			}
+		}
+		w.isSpillPrepared = true
```

Comment:
Do we need to set this to false in Close?

Reply:
> Do we need to set this to false in Close?

no, as we would re-create workers in `Open()`